src/ingest/src/main/scala/osmesa/ingest/util/Caching.scala

```diff
@@ -0,0 +1,73 @@
+package osmesa.ingest.util
+
+import com.amazonaws.services.s3.AmazonS3ClientBuilder
+import org.apache.spark.sql._
+import org.apache.log4j.{Level, Logger}
+
+import java.io.File
+
+
+trait Caching {
+  def orc(filename: String)(sparkjob: => DataFrame)(implicit ss: SparkSession): DataFrame
+}
+
+class S3Caching(cacheDir: String) extends Caching {
+  lazy val logger = Logger.getRootLogger()
+
+  private val s3client = AmazonS3ClientBuilder.defaultClient()
+
+  private val bucket = cacheDir.split("/")(2)
+
+  private def prefix(filename: String): String =
+    cacheDir.split("/").drop(3).mkString("/") + "/" + filename
+
+  private def fileUri(filename: String): String =
+    cacheDir + "/" + filename
+
+  private def s3exists(filename: String): Boolean =
+    s3client.doesObjectExist(bucket, prefix(filename) + "/_SUCCESS")
+
+  def orc(filename: String)(sparkjob: => DataFrame)(implicit ss: SparkSession): DataFrame = {
+    logger.warn("in cache func")
+    logger.warn("bucket: " ++ bucket)
+    logger.warn(prefix(filename))
+    logger.warn(fileUri(filename))
+    logger.warn("s3exists: " ++ s3exists(filename).toString)
+    val res = if (s3exists(filename)) {
```

Comment:
See https://github.com/azavea/osmesa/pull/47/files#diff-a2dba02322df99efde18635c62176f34 for some thoughts of preventing duplicate work.